src/main/java/it/gov/pagopa/reward/notification/service/suspension/UserSuspensionServiceImpl.java

```diff
@@ -66,4 +73,51 @@ public Mono<RewardSuspendedUser> suspend(String organizationId, String initiativ
     public Mono<Boolean> isNotSuspendedUser(String initiativeId, String userId) {
         return rewardsSuspendedUserRepository.existsById(RewardSuspendedUser.buildId(userId, initiativeId)).map(b -> !b);
     }
+
+    @Override
+    public Mono<RewardSuspendedUser> readmit(String organizationId, String initiativeId, String userId) {
+        log.info("[REWARD_NOTIFICATION][USER_READMISSION] Readmitting suspended user having id {} on initiative {}",
+                userId, initiativeId);
+
+        RewardSuspendedUser rollbackUser = new RewardSuspendedUser(userId, initiativeId, organizationId);
+
+        return PerformanceLogger.logTimingFinally("READMISSION",
+                notificationRuleRepository.findByInitiativeIdAndOrganizationId(initiativeId, organizationId)
+                        .flatMap(i -> rewardsSuspendedUserRepository.findByUserIdAndOrganizationIdAndInitiativeId(
+                                                userId,
+                                                organizationId,
+                                                initiativeId
+                                        )
+                                        .flatMap(u -> rewardsSuspendedUserRepository.delete(u)
+                                                .then(Mono.just(u))
+                                        )
+                                        .flatMap(u -> rewardsNotificationRepository.findByUserIdAndInitiativeIdAndStatus(u.getUserId(), u.getInitiativeId(), RewardNotificationStatus.SUSPENDED)
+                                                .flatMap(this::readmitNotifications)
+                                                .then(Mono.just(u))
+                                        )
+                                        .flatMap(u -> walletRestClient.readmit(u.getInitiativeId(), u.getUserId())
```

Comment:
could wallet readmit REST service return empty?

What if wallet doesn't know the current user as suspended? would it still return OK?